Fixed #22383 -- Added support for HTML5 required attribute on required form fields.

jdufresne opened the pull request:
<https://code.djangoproject.com/ticket/22383>

Same as PR #6341 but without the deprecation cycle.

timgraham commented on docs/ref/forms/api.txt:
```diff
@@ -692,6 +692,13 @@ This will take precedence over :attr:`Form.label_suffix
+.. versionadded:: 1.10
+
+When ``True``, required form fields will have the HTML attribute
+``required``. By default, ``use_required_attribute`` is ``True``.
```
Should explain how this works for formsets somewhere. Maybe here and elsewhere?

timgraham commented on docs/releases/1.10.txt:
```diff
@@ -250,6 +250,10 @@ Forms
   :mod:`django.contrib.staticfiles` if installed.
 
+* Form now has the attribute :attr:`~django.forms.Form.use_required_attribute`.
+  When set to ``True``, required form fields will have the HTML attribute
```
"When set to True (the default), ...." (and chop the next sentence)
Also add a mention in the backwards-incompatible changes section. Can mention "<form novalidate>" .

timgraham commented:
Please check the docs for rendered form examples that might need to be updated, e.g. `docs/topics/forms/index.txt`.

jdufresne commented:
I have addressed the concern. All additional feedback is welcome. Thanks.

> Should explain how this works for formsets somewhere. Maybe here and elsewhere?

Added some text to `ref/forms/api.txt ` and to `topics/forms/formsets.txt`. Please let me know what you think and if this addresses what you had in mind.

> Please check the docs for rendered form examples that might need to be updated, e.g. docs/topics/forms/index.txt.

Done. This resulted in changes to:

* `refs/forms/api.txt`
* `refs/forms/fields.txt`
*  `refs/forms/widgets.txt`
*  `topics/forms/index.txt`

timgraham commented:
merged in ec6121693f112ae33b653b4364e812722d2eb567, thanks!
